doc/services/shell/index.rst

```diff
@@ -97,6 +99,86 @@ considerably. For that cost, it will enable the line editing,
 `tab completion <tab-feature_>`_, and `history <history-feature_>`_
 features of the shell.
 
+USB CDC ACM Backend
+===================
```

Comment:
it looks like backend "headings" are a bit all over the place (see USB CDC, Telnet, RTT, ...) can you have a look and see if at a minimum you can't group USB CDC and Telnet together under the same heading (and maybe move RTT there too, if appropriate?)

Reply:
I've updated this branch attempting to gather all the already described backends. However, the list is not exhaustive and did not try to group using Shell as Logging backend: hopefully that's okay!